pkg/cloud/services/networking/port.go

```diff
@@ -141,6 +142,14 @@ func (s *Service) GetOrCreatePort(eventObject runtime.Object, clusterName string
 		FixedIPs:            fixedIPs,
 	}
 
+	if portOpts.QoSPolicyID != nil {
+		policyID := portOpts.QoSPolicyID
+		createOpts = policies.PortCreateOptsExt{
```

Comment:
Note that the policy has to exist before the port gets created. But since we have to provide an UUID to it, we expect the policy to exist. In the case of the policy can't be found (e.g. owned by another tenant), Neutron should return an API error where the policy can't be found.

Reply:
we have error check at line 170 below, so should be ok to avoid the check

Reply:
This is also consistent with what I've done in #994: if the user explicitly gives us a resource UUID we trust it without further validation. If it's wrong, a subsequent operation will fail and they can see the error about that.